numpy/ma/core.py

```diff
@@ -3105,15 +3105,18 @@ def __getitem__(self, indx):
         Return the item described by i, as a masked array.
 
         """
-        # 2016.01.15 -- v1.11.0
-        warnings.warn(
-            "Currently, slicing will try to return a view of the data," +
-            " but will return a copy of the mask. In the future, it will try" +
-            " to return both as views.",
-            FutureWarning
-        )
-
         dout = self.data[indx]
+
+        if (isinstance(dout, np.ndarray) and
+                np.may_share_memory(dout, self.data)):
```

Comment:
It should be fast, it just calculates the maximum (possible) memory bounds of both arrays and compares them. It has low specificity, but that does not matter here.
It makes the message here specific, but since the setitem one is not really specific at all, not sure how much it gives us. Will remove this code from this PR soon anyway, I had had a faint hope that it would reduce the mass of warnings to something acceptable (for more specific suppression), but that did not happen anyway.